scripts/dex-router/src/router/jupiter.ts

```diff
@@ -0,0 +1,88 @@
+import { Router, RouterOutput } from "../types";
+import { PublicKey, TransactionInstruction } from "@solana/web3.js";
+import {
+  createJupiterApiClient,
+  Instruction as JupiterInstruction,
+} from "@jup-ag/api";
+
+const maxAccounts = 20;
+const maxAutoSlippageBps = 50;
+
+export class JupiterRouter implements Router {
+  private chainId: string;
+  private executor: PublicKey;
+  private jupiterClient = createJupiterApiClient();
+
+  constructor(chainId: string, executor: PublicKey) {
+    this.chainId = chainId;
+    this.executor = executor;
+  }
+
+  async route(
+    tokenIn: PublicKey,
+    tokenOut: PublicKey,
+    amountIn: bigint
+  ): Promise<RouterOutput> {
+    if (!["mainnet-beta-solana", "development-solana"].includes(this.chainId)) {
+      throw new Error("Jupiter error: chain id not supported");
+    }
+
+    const quoteResponse = await this.jupiterClient.quoteGet({
+      inputMint: tokenIn.toBase58(),
+      outputMint: tokenOut.toBase58(),
+      amount: Number(amountIn),
+      autoSlippage: true,
+      maxAutoSlippageBps,
+      maxAccounts: maxAccounts,
+    });
+
+    const instructions = await this.jupiterClient.swapInstructionsPost({
+      swapRequest: {
+        userPublicKey: this.executor.toBase58(),
+        quoteResponse,
+      },
+    });
+
+    const {
+      computeBudgetInstructions,
+      setupInstructions,
+      swapInstruction,
+      addressLookupTableAddresses,
+    } = instructions;
+
+    const ixsComputeBudget = computeBudgetInstructions.map((ix) =>
```

Comment:
is it possible for the compute budget to be low and we run out of compute budget because of the other instructions we have included in the tx?

Reply:
in theory yes, but for now the returned compute unit limit is always 1.4 million bc we aren't using the dynamic CU limit option in Jupiter request

we can either remove this instruction or have a minimum CU limit here or add CU units based on the compute unit reqs of `SubmitBid` and the limo ix

Reply:
cool, so let's just add a comment here mentioning this.